dashboard/src/components/userSettings/UserSettingsDialog.tsx

```diff
@@ -86,23 +159,28 @@ export default function UserSettingsDialog({ open, onOpenChange }: UserSettingsD
         component: UserDangerZoneSettings,
       },
     ],
-    [isFeatureFlagEnabled],
+    [isFeatureFlagEnabled, tTabs],
   );
 
   const availableTabs = USER_SETTINGS_TABS.filter((tab) => !tab.disabled);
   const [activeTab, setActiveTab] = useState(availableTabs[0].id);
-  const [formData, setFormData] = useState<UserSettingsUpdate>({});
+  const [formData, setFormData] = useState<UserSettingsUpdate>({ ...settings });
   const { refreshSession } = useSessionRefresh();
   const isFormChanged = useIsChanged(formData, settings);
 
   useEffect(() => {
-    if (settings) {
+    if (open) {
       setFormData({ ...settings });
     }
-  }, [settings]);
+  }, [settings, open]);
+
+  useEffect(() => {
+    return () => restore();
+  }, [restore]);
```

Comment:
This is an anti-pattern 😄 
I just tested it having removed this, and it seems to fully work as expected. I think it's due to also being in the `handleCloseDialog`
```suggestion
```